rotkehlchen/chain/evm/decoding/monerium/decoder.py

```diff
@@ -0,0 +1,99 @@
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.constants import ERC20_OR_ERC721_TRANSFER
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+from .constants import BURN_MONERIUM_SIGNATURE, CPT_MONERIUM
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+class MoneriumCommonDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            native_asset: Asset,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.native_asset = native_asset
+        self.eure_token = native_asset.resolve_to_evm_token()
+
+    def _decode_mint_and_burn(self, context: DecoderContext) -> DecodingOutput:
+        """Decode mint and burn events for monerium"""
+
+        if context.tx_log.topics[0] != ERC20_OR_ERC721_TRANSFER:
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = hex_or_bytes_to_address(context.all_logs[0].topics[1])
+        to_address = hex_or_bytes_to_address(context.all_logs[0].topics[2])
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data)
+        amount = token_normalized_value_decimals(amount_raw, self.eure_token.decimals)
+
+        event = None
+        if from_address == ZERO_ADDRESS:
+            # Create a mint event
+            mint_event = self.base.make_event_from_transaction(
+                transaction=context.transaction,
+                tx_log=context.tx_log,
+                event_type=HistoryEventType.RECEIVE,
+                event_subtype=HistoryEventSubType.NONE,
+                asset=self.native_asset,
+                balance=Balance(amount=amount),
+                location_label=to_address,
+                notes=f'Mint {amount} {self.eure_token.symbol}',
+                counterparty=CPT_MONERIUM,
+            )
+            event = mint_event
+
```

Comment:
you don't need mint_event later so you can just do

```suggestion
            # Create a mint event
            event = self.base.make_event_from_transaction(
                transaction=context.transaction,
                tx_log=context.tx_log,
                event_type=HistoryEventType.RECEIVE,
                event_subtype=HistoryEventSubType.NONE,
                asset=self.native_asset,
                balance=Balance(amount=amount),
                location_label=to_address,
                notes=f'Mint {amount} {self.eure_token.symbol}',
                counterparty=CPT_MONERIUM,
            )
```